src/lib/blueprint.ts

```diff
@@ -1124,6 +1134,19 @@ const validateRoutePath = (
   return routePath
 }
 
+// Store the property groups for each resource during processing
+const resourcePropertyGroups: Record<string, Record<string, PropertyGroup>> = {}
```

Comment:
We can't use a global cache like this in a library. If you need something like this, you can pass it through as a parameter.

Reply:
34cc4890bcf1a5fc9b0913b0ff82dc48b56f2f47